vernac/comTactic.mli

```diff
@@ -0,0 +1,51 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+(** Tactic interpreters have to register their interpretation function *)
+type 'a tactic_interpreter
+type interpretable = I : 'a tactic_interpreter * 'a -> interpretable
+
+(** ['a] should be marshallable if ever used with [par:] *)
+val register_tactic_interpreter :
+  string -> ('a -> unit Proofview.tactic) -> 'a tactic_interpreter
+
+(** Entry point for toplevel tactic expression execution. It calls Proof.solve
+    after having interpreted the tactic, and after the tactic runs it
+    unfocus as much as needed to put a goal under focus. *)
+val solve :
+  pstate:Declare.Proof.t ->
+  Goal_select.t ->
+  info:int option ->
+  interpretable ->
+  with_end_tac:bool ->
+  Declare.Proof.t
+
+(** [par: tac] runs tac on all goals, possibly in parallel using a worker pool.
+    If tac is [solve abstract tac1], then [solve] and [abstract] are passed
+    explicitly to the solver since they are of interest to the master (the
+    syntax of ltac is not visible here).
+    In particular if [solve] is true and one job fails master may cancel the
+    other ones. When [abstract] is true, it is up to master to opacify the
```

Comment:
Why is master not allowed to cancel without solve?
Also it should be clearer that for abstract it is removed from the passed `tac` and indicated solely through the flag, and for solve it is left in the `tac` and duplicated in the flag.

Reply:
> Why is master not allowed to cancel without solve?

Hum, I think that my understanding of `all:` back then was wrong. Apparently the code can be simplified: no need to peel `solve` off, pass it as `~solve` and finally no need to have an if in the cancelling code.

I'm going to fix it right now.